internal/cri/opts/spec_linux_opts.go

```diff
@@ -40,7 +40,7 @@ import (
 )
 
 // WithMounts sorts and adds runtime and CRI mounts to the spec
-func WithMounts(osi osinterface.OS, config *runtime.ContainerConfig, extra []*runtime.Mount, mountLabel string, handler *runtime.RuntimeHandler) oci.SpecOpts {
+func WithMounts(osi osinterface.OS, config *runtime.ContainerConfig, extra []*runtime.Mount, mountLabel string, handler *runtime.RuntimeHandler, cgroupWritable bool) oci.SpecOpts {
```

Comment:
This is an exported function, although it's in an `internal` pkg. I'm not sure if it's okay to change its interface.

Another approach is adding another similar func `WithMountsCgroupWritable`, and share the main logic between the two:

```golang 

func WithMounts(...) {
  return withMounts(..., false)
}

func WithMountsCgroupWritable(...) {
  return withMounts(..., true)
}

func withMounts(..., cgroupWritable bool) {
  // move main logic here
}
```